cpp/src/arrow/python/arrow_to_python.cc

```diff
@@ -195,6 +195,7 @@ Status ReadSerializedPythonSequence(std::shared_ptr<io::RandomAccessFile> src,
   RETURN_NOT_OK(ipc::RecordBatchStreamReader::Open(src, &reader));
   RETURN_NOT_OK(reader->ReadNextRecordBatch(batch_out));
   RETURN_NOT_OK(src->Tell(&offset));
+  offset += 4; // TODO(pcm): Why is this neccessary?
```

Comment:
There's a 0 EOS in the stream: https://github.com/apache/arrow/blob/master/cpp/src/arrow/ipc/writer.cc#L646